src/snyk/base/services/authenticationService.ts

```diff
@@ -91,22 +92,32 @@ export class AuthenticationService implements IAuthenticationService {
     await this.contextService.setContext(SNYK_CONTEXT.LOGGEDIN, false);
   }
 
-  async setToken(): Promise<void> {
-    const token = await this.window.showInputBox({
-      placeHolder: 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx',
-      password: true,
-      validateInput: token => {
-        const valid = uuidValidate(token);
-        if (!valid) {
-          return 'The entered token has an invalid format.';
-        }
-      },
-    });
+  async setToken(passedToken?: string): Promise<void> {
+    const invlaidTokenMsg = 'The entered token has an invalid format.';
```

Comment:
This method has UI logic as well as business logic mixed. How about we keep them separate as it was before?

Reply:
👍 I think even not using this method at all for updating the token as there isn't any shared logic as originally anticipated; I essentially just added branching logic to a method that didn't need it.